src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -255,54 +255,59 @@ public static function get_user_data_from_wp_global_styles( $theme, $create_post
 		}
 		$user_cpt         = array();
 		$post_type_filter = 'wp_global_styles';
+		$stylesheet       = $theme->get_stylesheet();
 		$args             = array(
-			'numberposts' => 1,
-			'orderby'     => 'date',
-			'order'       => 'desc',
-			'post_type'   => $post_type_filter,
-			'post_status' => $post_status_filter,
-			'tax_query'   => array(
+			'posts_per_page'      => 1,
+			'orderby'             => 'post_date',
+			'order'               => 'desc',
+			'post_type'           => $post_type_filter,
+			'post_status'         => $post_status_filter,
+			'ignore_sticky_posts' => true,
+			'no_found_rows'       => true,
+			'tax_query'           => array(
 				array(
 					'taxonomy' => 'wp_theme',
 					'field'    => 'name',
-					'terms'    => $theme->get_stylesheet(),
+					'terms'    => $stylesheet,
 				),
 			),
 		);
 
 		$cache_key = sprintf( 'wp_global_styles_%s', md5( serialize( $args ) ) );
-		$post_id   = wp_cache_get( $cache_key );
-
-		if ( (int) $post_id > 0 ) {
-			return get_post( $post_id, ARRAY_A );
-		}
-
+		$post_id   = (int) get_transient( $cache_key );
 		// Special case: '-1' is a results not found.
 		if ( -1 === $post_id && ! $create_post ) {
 			return $user_cpt;
 		}
 
-		$recent_posts = wp_get_recent_posts( $args );
-		if ( is_array( $recent_posts ) && ( count( $recent_posts ) === 1 ) ) {
-			$user_cpt = $recent_posts[0];
+		if ( $post_id > 0 && in_array( get_post_status( $post_id ), (array) $post_status_filter, true ) ) {
+			return get_post( $post_id, ARRAY_A );
+		}
+
+		$gs_query     = new WP_Query();
+		$recent_posts = $gs_query->query( $args );
```

Comment:
Per [56448](https://core.trac.wordpress.org/ticket/56448#comment:6) we should use meaningful names for the variables.
```suggestion
		$global_style_query = new WP_Query();
		$recent_posts       = $global_style_query->query( $args );
```